source/agora/test/QuorumShuffle.d

```diff
@@ -42,140 +43,81 @@ unittest
 
     auto nodes = network.clients;
 
-    nodes.enumerate.each!((idx, node) =>
-        retryFor(node.getBlockHeight() == 2, 5.seconds,
-            format("Node %s has block height %s. Expected: %s",
-                idx, node.getBlockHeight(), 2)));
+    Height block_height = Height(0);
 
     const keys = WK.Keys.byRange.map!(kp => kp.address).take(6).array;
 
-    // 0 because the shuffle occured at genesis, not at block height 2
-    const quorums_1 = nodes[0].getExpectedQuorums(keys, Height(0));
-    // writeln(quorums_1);  // uncomment to check
-    nodes.enumerate.each!((idx, node) =>
-        retryFor(node.getQuorumConfig() == quorums_1[idx], 5.seconds,
-            format("Node %s has quorum config %s. Expected: %s",
-                idx, node.getQuorumConfig(), quorums_1[idx])));
-
     // check that the preimages were revealed before we trigger block creation
     // note: this is a workaround. A block should not be accepted if there
     // are no preimages for this height - however currently block signatures
     // are not implemented yet (#365)
-    const enrollments = network.blocks[0].header.enrollments;
-    void checkDistance (uint distance)
+    void checkDistance (Height height)
     {
-        nodes.each!(node =>
+        auto enrolled_height = Height(GenesisCycle * ((height.value - 1) / GenesisCycle));
+        auto required_distance = (height - enrolled_height - 1) % GenesisCycle;
+        log.trace(format!"check distance is %s for generating block at height %s with enrollments at height %s"(required_distance, height, enrolled_height));
+        auto enrollments = nodes[0].getBlocksFrom(enrolled_height, 1)[0].header.enrollments;
+        nodes.enumerate.each!((idx, node) =>
             enrollments.each!(enr =>
-                retryFor(node.getPreimage(enr.utxo_key).distance >= distance,
-                    5.seconds)));
+                retryFor(node.getPreimage(enr.utxo_key).distance >= required_distance, 5.seconds,
+                    format!"node #%s has preimage distance %s not %s as expected"
+                        (idx, node.getPreimage(enr.utxo_key).distance, required_distance))));
     }
 
-    checkDistance(3);
-
-    const(Block)[] blocks = [network.blocks[$ - 1]];
-    void makeBlock (ulong height)
+    void makeBlock ()
     {
-        blocks[$ - 1].spendable.each!(txb =>
+        nodes[0].getBlocksFrom(block_height++, 1)[0].spendable.takeExactly(8).each!(txb =>
             nodes[0].putTransaction(txb.sign()));
-        network.expectBlock(Height(height));
-        retryFor((blocks = nodes[0].getBlocksFrom(height, 1)).length == 1, 5.seconds,
-                format("Node 0 getBlocksFrom(%s, 1) failed", height));
+        log.trace(format!"make block %s"(block_height));
```

Comment:
Why not just `log.trace("make block {}", block_height);`? It seems overkill to use templated `format` here (remember that each template instantiation will cause code bloat and slow down compilation).

Reply:
Updated